fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStore.java

```diff
@@ -2477,12 +2478,12 @@ private boolean areAllRecordTypesSince(@Nullable Collection<RecordType> recordTy
         }
         final boolean metaDataVersionChanged = oldMetaDataVersion != newMetaDataVersion;
 
-        if (!formatVersionChanged && !metaDataVersionChanged) {
+        final boolean newStore = oldFormatVersion == 0;
+        if (!newStore && !formatVersionChanged && !metaDataVersionChanged) {
```

Comment:
I guess I'm trying to understand the mechanism here.

Prior to this change, if there is a new store where the `metaDataVersion` of a new store is 0, then `metaDataVersionChanged` is `false` because `oldMetaDataVersion == info.getMetaDataversion() == 0 == newMetaDataVersion`. But shouldn't `formatVersionChanged` be `true`, as `oldFormatVersion == info.getFormatVersion() == 0 != newFormatVersion`?

Reply:
The idea was to mirror the conditions inside the logging to be a general "nothing to do" check. But there's no real reason not to keep it simple (and the way it was).

Reply:
Oh, I think I misunderstood the problem. I thought the issue was that store header wouldn't get saved *at all* if the meta-data version was zero, and I was confused how this changed things (assuming a zero to non-zero format version). But if I'm reading the issue correctly, the problem was actually that the store header was saved, but the meta-data version was left blank rather than set to something (in this case, zero). Okay, this makes more sense now.